interface/forms/procedure_order/common.php

```diff
@@ -0,0 +1,1287 @@
+                } else if (exception === 'timeout') {
+                    alert('Time out error.');
+                } else if (exception === 'abort') {
+                    alert('Ajax request aborted.');
+                } else {
+                    alert('Uncaught Error.\n' + jqXHR.responseText);
+                }
+            }
+        })
+        return false;
+    });
+    <?php if ($row['date_transmitted']) { ?>
+    $("#summary").collapse("toggle");
+    <?php } ?>
+});
+
+function getDetails(e, id) {
+    top.restoreSession();
+    let f = document.forms[0];
+    let codeattr = 'form_proc_code[' + id + ']';
+    let codetitle = 'form_proc_type_desc[' + id + ']';
+    let code = f[codeattr].value;
+    let url = top.webroot_url + "/interface/procedure_tools/libs/labs_ajax.php";
+    url += "?action=code_detail)&code=" + encodeURIComponent(code) +
+        "&csrf_token_form=" + <?php echo js_url(CsrfUtils::collectCsrfToken()); ?>;
+    let title = <?php echo xlj("Test") ?> + ": " + code + " " + f[codetitle].value;
+    dlgopen(url, 'details', 'modal-md', 200, '', title, {
+        buttons: [
+            {text: '<?php echo xlt('Got It'); ?>', close: true, style: 'secondary btn-sm'}
+        ]
+    });
+}
+
+function initForm(reload = false) {
+    if (reload) {
+        location.reload();
+    }
+    $(".ammon").hide();
+    $(".quest").hide();
+    $(".labcorp").hide();
+    $(".clarity").hide();
+    $(".defaultProcedure").hide();
+
+    if (currentLab === 'ammon') {
+        $(".ammon").show();
+    } else if (currentLab === 'quest') {
+        $(".quest").show();
+    } else if (currentLab === 'labcorp') {
+        $(".labcorp").show();
+    } else if (currentLab === 'clarity') {
+        $(".clarity").show();
+    } else if (currentLab === '') {
+        $(".defaultProcedure").show();
+    }
+}
+
+function createLabels(e) {
+    e.preventDefault();
+    let prmt = <?php echo js_escape(xlt("How many sets of specimen labels to create?") .
+        "\n" . xlt("Each test in order gets a label.")); ?>;
+    let count = prompt(prmt, '1');
+    if (!count) return false;
+    let tarray = "";
+    let f = document.forms[0];
+    let transport = '';
+    let i = 0;
+    for (; f['form_transport[' + i + ']']; ++i) {
+        transport = f['form_transport[' + i + ']'].value;
+        transport = transport > '' ? transport : 'none';
+        tarray += transport + ";";
+    }
+    let printer = 'file';
+    let acctid = '<?php echo $gbl_client_acct ?>';
+    let order = f.id.value;
+    let patient = "<?php echo $patient['lname'] . ', ' . $patient['fname'] . ' ' . $patient['mname'] ?>";
+    let pid = "<?php echo $patient['pid']  ?>";
+    let url = top.webroot_url + "/interface/procedure_tools/libs/labs_ajax.php";
+    // this escapes above
+    let uri = encodeURI("?action=print_labels&count=" + count + "&order=" + order + "&pid=" + pid +
+        "&acctid=" + acctid + "&patient=" + patient + "&specimen=" + tarray) +
```

Comment:
```
    let uri = encodeURI("?action=print_labels&count=" + encodeURIComponent(count) + "&order=" + encodeURIComponent(order) + "&pid=" + encodeURIComponent(pid) +
        "&acctid=" + encodeURIComponent(acctid) + "&patient=" + encodeURIComponent(patient) + "&specimen=" + encodeURIComponent(tarray)) +
```
(the `tarray)` string is throwing me off (i kept the `)` character from it above))

Reply:
oh, i think i see the issue here. you are using encodeURI on entire string. Rec not doing that and instead doing encodeURIComponent on the individual elements (so that special characters in variables such as ?,=,& then won't break things)